internal/security/proxy/service.go

```diff
@@ -0,0 +1,332 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.1.0
+ *******************************************************************************/
+package proxy
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/dghubble/sling"
+	"net/http"
+)
+
+type Service struct {
+	Connect    Requestor
+	CertCfg    CertConfig
+	ServiceCfg ServiceConfig
+}
+
+type ServiceConfig interface {
+	GetProxyAuthMethod() string
+	GetProxyAuthTTL() int
+	GetProxyAuthResource() string
+	GetProxyACLName() string
+	GetProxyACLWhiteList() string
+	GetSecretSvcSNIS() string
+	GetEdgeXSvcs() map[string]service
+}
+
+func (s *Service) CheckProxyServiceStatus() error {
+	return s.checkServiceStatus(s.Connect.GetProxyBaseURL())
+}
+
+func (s *Service) CheckSecretServiceStatus() error {
+	return s.checkServiceStatus(s.Connect.GetSecretSvcBaseURL())
+}
+
+func (s *Service) checkServiceStatus(path string) error {
+	req, err := sling.New().Get(path).Request()
+	resp, err := s.Connect.GetHTTPClient().Do(req)
+	if err != nil {
+		e := fmt.Sprintf("the status of service on %s is unknown, the initialization is terminated", path)
+		return errors.New(e)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		lc.Info(fmt.Sprintf("the service on %s is up successfully", path))
+		break
+	default:
+		e := fmt.Sprintf("the service on %s is down", path)
+		return errors.New(e)
```

Comment:
fixed.

Reply:
how did you fix it though?

Reply:
we have lc.Error(e) in the spot and return error as well. I would prefer to have error logged at the place where it happens in case the call didn't log the error for some reason.

Reply:
I think this should be discussed in the WG this week